### PR TITLE
Add max telemetry bytes

### PR DESCRIPTION
**Pull request by anchitj**

### CODE REVIEW

File: src/rdkafka_telemetry.c [hunk 1]
```diff
@@ -225,6 +226,14 @@ static void rd_kafka_send_push_telemetry(rd_kafka_t *rk,
         // TODO: Cycle through compression types
         rd_kafka_compression_t compression_type = RD_KAFKA_COMPRESSION_GZIP;
 
+        if (metrics_payload_size > rk->rk_telemetry.telemetry_max_bytes) {
+                rd_kafka_log(
+                    rk, LOG_WARNING, "TELEMETRY",
+                    "Metrics payload size %" PRIdsz
+                    " exceeds telemetry_max_bytes %d specified by the broker.",
```
Comment (milindl) [inline src/rdkafka_telemetry.c:233]: The format string will PRId32 for int32

File: src/rdkafka_telemetry.c [hunk 2]
```diff
@@ -271,6 +280,19 @@ void rd_kafka_handle_push_telemetry(rd_kafka_t *rk, rd_kafka_resp_err_t err) {
                 rd_kafka_dbg(rk, TELEMETRY, "PUSHERR",
                              "PushTelemetryRequest failed: %s",
                              rd_kafka_err2str(err));
+                // Non-retriable errors
+                // TODO: What to do for TELEMETRY_TOO_LARGE?
```
Comment (milindl) [inline src/rdkafka_telemetry.c:284]: I guess we can just add a comment. Or at max a log (even though we will have logged a warning already), based on the discussion with the Java clients team, we need to do nothing special.

**Review by milindl — APPROVED** [review]
Looks okay - just a bunch of minor stuff

File: src/rdkafka_telemetry.c (hunk 1 above)
Comment (anchitj) [reply src/rdkafka_telemetry.c:233]: Updated

File: src/rdkafka_telemetry.c (hunk 2 above)
Comment (anchitj) [reply src/rdkafka_telemetry.c]: Removed the comment

**Comment by milindl** [comment]
LGTM

**Pull request merged**